File: rently-backend/src/main/java/com/rently/rentlybackend/dto/response/PropertyResponse.java
package com.rently.rentlybackend.dto.response;

import java.math.BigDecimal;

public record PropertyResponse(
        String id,
        String ownerId,
        String propertyType,
        String name,
        String description,
        Integer maxNumberOfGuests,
        BigDecimal pricePerNight,
        boolean approved,
        Double averageRate,
        Address address
) {
    public record Address(
            String country,
            String city,
            String street,
            String houseNumber,
            String apartmentNumber,
            String postalCode
    ) {}
}

File: rently-frontend/src/types/PropertyResponse.ts
export type PropertyResponse  = {
  id: string;
  ownerId: string;
  propertyType: string;
  name: string;
  description: string;
  maxNumberOfGuests: number;
  pricePerNight: number;
  approved: boolean;
  averageRate: number;
  images: string[];
  address: {
    country: string;
    city: string;
    street: string;
    houseNumber: string;
    apartmentNumber: string;
    postalCode: string;
  };
}
